.github/workflows/node.js.yml

```diff
@@ -0,0 +1,95 @@
+name: Node.js CI
+
+on:
+  push:
+    branches:
+      - jp-gha-conversion
+      - staging
+      # - master
+  pull_request:
+    branches:
+      - staging
+      # - master
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Checks & Build Test
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v3
+
+      - name: Node.js setup
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16.x'
+
+      - name: 📦 Install dependencies
+        run: yarn install
+
+      - name: 💅 Prettier check
+        run: yarn prettier:check
+
+      - name: 💨 Lint check
+        run: yarn lint
+
+      - name: 🏗️ Build app
+        run: yarn build --prod
+
+  deploy:
+    name: Deploy
+    if: ${{ github.event_name == 'push' && (github.ref == 'refs/heads/master' || github.ref == 'refs/heads/staging' || github.ref == 'refs/heads/jp-gha-conversion')}}
+    runs-on: ubuntu-latest
+    environment:
+      name: ${{ (github.ref == 'refs/heads/master' && 'prod') || 'stage' }}
+    needs: build
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v3
+
+      - name: Install AWSCLI
+        run: pip install --user awscli
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: us-east-1
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE_ARN }}
```

Comment:
Personally I would move this step right before the `Deploy to s3` step, that will keep the scope of what steps actually have access to AWS credentials more limited